### DISCUSSION TITLE
Module did not self-register: '../node_modules/canvas/build/Release/canvas.node'

### DISCUSSION THREAD

**Discussion by polly3d** [nuxt/nuxt#17901] [Questions]
I use fabric.js and it needs [cavans](https://github.com/Automattic/node-canvas) library. It runs well when starts the project. But failed after saving files or refreshing the page. I known that canvas doesn't support worker_threads from [this issue](https://github.com/Automattic/node-canvas/issues/1536).  

Is there a way to let the compiler doesn't use the worker_threads?

**Comment by and2** [comment]
Unless you need to use the `canvas` library, a simple solution would be to exclude the dependancy from the fabric library by adding `-browser` to the fabric version in `package.json` as per [this thread](https://github.com/fabricjs/fabric.js/pull/5786). 

`"fabric": "5.2.1-browser"`
